arbnode/transaction_streamer.go

```diff
@@ -205,6 +213,10 @@ func (s *TransactionStreamer) SetInboxReaders(inboxReader *InboxReader, delayedB
 	}
 	s.inboxReader = inboxReader
 	s.delayedBridge = delayedBridge
+
+	if s.syncMonitor != nil && inboxReader != nil && inboxReader.tracker != nil {
+		s.broadcastChecker = NewBroadcastSyncChecker(s.syncMonitor, inboxReader.tracker)
```

Comment:
This is also introducing a circular dependency:
TransactionStreamer -> BroadcastSyncChecker -> SyncMonitor -> TransactionStreamer

This also indicates to me that all of them are better placed in a single component.
How about moving BroadcastSyncChecker logic to TransactionStreamer too?